pkg/machinehealthcheck/machinehealthcheck.go

```diff
@@ -101,24 +88,95 @@ var _ = Describe("[Feature:MachineHealthCheck] MachineHealthCheck", func() {
 					Timeout: "1s",
 				},
 			},
+			MaxUnhealthy: &maxAllowedUnhealthy,
 		}
 
 		mhc, err := framework.CreateMHC(client, mhcParams)
 		Expect(err).ToNot(HaveOccurred())
 		Expect(mhc).ToNot(BeNil())
 
 		By("Verifying the matching Machine is deleted")
-		framework.WaitForMachineDelete(client, machine)
+		framework.WaitForMachinesDeleted(client, machine)
 
 		By("Verifying the MachineSet recovers")
 		framework.WaitForMachineSet(client, machineSet.GetName())
 
 		By("Deleting the MachineHealthCheck resource")
 		err = client.Delete(context.Background(), mhc)
 		Expect(err).ToNot(HaveOccurred())
+
+		By("Deleting the new MachineSet")
+		err = client.Delete(context.Background(), machineSet)
+		Expect(err).ToNot(HaveOccurred())
+
+		framework.WaitForMachineSetDelete(client, machineSet)
 	})
 
-	// It("should respect maxUnhealthy", func() {
-	// 	// TODO(bison): This.
-	// })
+	It("should respect maxUnhealthy", func() {
+		machineSetParams = framework.MachineSetParams{
+			Name:         msName,
+			Replicas:     3,
+			ProviderSpec: providerSpec,
+			Labels: map[string]string{
+				"mhc.framework.openshift.io": msName,
+				framework.ClusterKey:         clusterName,
+			},
+		}
+
+		By("Creating a new MachineSet")
```

Comment:
Creating and deleting machineset is done in `BeforeEach()` and `AfterEach()`

Reply:
https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-api-actuator-pkg/160/pull-ci-openshift-cluster-api-actuator-pkg-master-e2e-gcp-operator/195 What else could cause this, but a double create call in a parallel execution of `BeforeEach` with a create call. There is definitely a race condition somewhere. Something like `BeforeAll`/`AfterAll` would be perfect

Reply:
Realized that it is no longer needed.